zerver/views/auth.py

```diff
@@ -772,7 +817,10 @@ def login_page(
         # HttpResponse. See django.template.response.SimpleTemplateResponse,
         # https://github.com/django/django/blob/master/django/template/response.py#L19.
         update_login_page_context(request, template_response.context_data)
-
+    else:
```

Comment:
A simple `else` here seems extremely unsafe. You're assuming anything else than `SimpleTemplateResponse` means the user authenticated successfully. Even if that's true right now, it might not be so for example if Django just change the base class to something else. Or if there's possibility of something else being returned in certain cases of the user failing to auth. We need to find out what exactly the expected `template_response` is in case of success and verify it here.

Reply:
It is just a `HttpResponseRedirect`(added a print, also verified [here](https://github.com/django/django/blob/3f2821af6bc48fa8e7970c1ce27bc54c3172545e/django/contrib/auth/views.py#L93)) not sure if it'd be safer but it's better than the current code. So added that check too.

Reply:
A stronger check here would be `request.user.is_authenticated`. So changed it check for that and continue to `confirm_email_change`.

Reply:
Great, `request.user.is_authenticated` seems like exactly the kind of thing we want. Mere fact of returning `HttpResponseRedirect` wouldn't inspire me with confidence that the user successfully authenticated.